emukit/quadrature/kernels/integration_measures.py

```diff
@@ -0,0 +1,147 @@
+        bounds_lower = np.array([b[0] for b in self.bounds])
+        bounds_upper = np.array([b[1] for b in self.bounds])
+        inside_lower = 1 - (x < bounds_lower)
+        inside_upper = 1 - (x > bounds_upper)
+        inside_upper_lower = (inside_lower * inside_upper).sum(axis=1) == x.shape[1]
+        return inside_upper_lower * self.density
+
+    def get_box(self) -> List[Tuple[float, float]]:
+        """
+        Meaningful box-bounds around the measure. Outside this box, the measure should be virtually zero.
+
+        :return: box in which the measure lies. List of D tuples, where D is the dimensionality and the tuples contain
+        the lower and upper bounds of the box i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+        """
+        return self.bounds
+
+
+class IsotropicGaussianMeasure(IntegrationMeasure):
+    """The isotropic Gaussian measure"""
+
+    def __init__(self, mean: np.ndarray, variance: float):
+        """
+        :param mean: the mean of the Gaussian, shape (dim, )
+        :param variance: the scalar variance of the isotropic covariance matrix of the Gaussian.
+        """
+        super().__init__('GaussianMeasure')
+        # check mean
+        if not isinstance(mean, np.ndarray):
+            raise TypeError('Mean must be of type numpy.ndarray, ' + str(type(mean)) + ' given.')
+
+        if mean.ndim != 1:
+            raise ValueError('Dimension of mean must be 1, dimension ' + str(mean.ndim) + ' given.')
+
+        # check covariance
+        if not isinstance(variance, float):
+            raise TypeError('Variance must be of type float, ' + str(type(variance)) + ' given.')
+
+        if not variance > 0:
+            raise ValueError('Variance must be positive, current value is ', variance, '.')
+
+        self.mean = mean
+        self.variance = variance
+        self.dim = mean.shape[0]
+
+    @property
+    def full_covariance_matrix(self):
+        return self.variance * np.eye(self.dim)
+
+    def compute_density(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the density at point x
+        :param x: points at which density is computes, shape (num_points, dim)
+        :return: the density at x, shape (num_points, )
+        """
+        factor = (2 * np.pi * self.variance) ** (self.dim / 2)
+        scaled_diff = (x - self.mean) / (np.sqrt(2 * self.variance))
+        return np.exp(- np.sum(scaled_diff ** 2, axis=1)) / factor
+
+    def get_box(self) -> List[Tuple[float, float]]:
+        """
+        Meaningful box-bounds around the measure. Outside this box, the measure should be virtually zero.
+
+        :return: box in which the measure lies. List of D tuples, where D is the dimensionality and the tuples contain
+        the lower and upper bounds of the box i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+        """
+        # Note: the factor 10 is pretty arbitrary here. Investigate if we can find a formula that adapts to
```

Comment:
Just curious if that actually matters much what factor is. Do we just need to pick something large enough so that it is (almost) guaranteed there is no data outside?

Reply:
It is the box that restricts the measure to a reasonable finite box, i,e., outside the box the density should be pretty much zero. 10 std is quite a large box for a Gaussian, so it is pretty conservative. I changed the comment to the one below.

"Note: the factor 10 is somewhat arbitrary but well motivated. If this method is used to get a box for data-collection, the box will be 2x 10 standard deviations wide in all directions, centered around the mean. Outside the box the density is virtually zero. "

Reply:
clear for me now, thanks